overlord/hookstate/ctlcmd/mount.go

```diff
@@ -219,20 +218,10 @@ func (m *mountCommand) Execute([]string) error {
 	}
 
 	sysd := systemd.New(systemd.SystemMode, nil)
-	name, err := m.createMountUnit(sysd)
+	_, err = m.ensureMount(sysd)
 	if err != nil {
 		return fmt.Errorf("cannot create mount unit: %v", err)
 	}
 
-	if err := sysd.Start([]string{name}); err != nil {
-		// There's no point in keeping the mount unit if it doesn't work. Even
-		// if the problem is transient, the next invocation of "snapctl mount"
-		// will create a new unit anyway.
-		if err := sysd.RemoveMountUnitFile(m.Positional.Where); err != nil {
```

Comment:
should we mode this under the error case of the ensureMount? it still makes a bit of sense not leave around a mount unit created on behalf of a snap that didn't work

Reply:
Done.

But there are other cases. If the mount existed, but was different. If we make a broken mount and it fails to reload, then we will unmount and delete the unit. In theory we would want to replace the unit by the previous version rather than removing it.